File: string/string1.java
import java.util.*;

public class string1{

public static void main(String args[]){
// duplicate();


// String str="abcd";
// SSfun(str);

}


public static void duplicate(){

    StringBuilder sb= new StringBuilder();    
    String str="AAABCDDDEFGH";

    for(int i=1;i<str.length()-1;i++){
        char pch= str.charAt(i-1);
        char ch=str.charAt(i); 
        if(pch!=ch){
            sb.append(pch);
            }
        }
        char lch=str.charAt(str.length()-1);
        sb.append(lch);
        System.out.println(sb.toString());
    }


public static void SSfun(String str){
    for(int i=0;i<str.length();i++){
        for(int j=i+1;j<=str.length();j++){
            System.out.println(str.substring(i,j));
        }
    }
    }
}

File: string/string1.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
// int counter=1;
int duplicate()
{
    string str = "AAABCDDDEFGH";
    int counter = 1;
    for (int i = 1; i < str.length() - 1; i++)
    {
        char ch = str[i];
        char pch = str[i - 1];
        if (pch != ch)
        {
            cout << pch << " ";
            if (counter > 1)
                cout << counter << " ";
            counter = 1;
        }
        else
        {
            counter++;
        }
    }
    char lch = str[str.length() - 1];
    cout << lch;
}

int main()
{
    // duplicate();
    return 0;
}